packages/ckeditor5/src/plugin.js

```diff
@@ -414,40 +471,49 @@ export default class MathType extends Plugin {
       "get",
       (e) => {
         let output = e.return;
-        // This line cleans all the semantics stuff, including the handwritten data points and returns the MathML IF there is any.
-        // For text or latex formulas, it returns the original output.
-        e.return = MathML.removeSemantics(output, "application/json");
+        const parsedResult = Parser.endParse(output);
+
+        // Cleans all the semantics tag for safexml
+        // including the handwritten data points
+        e.return = MathML.removeSafeXMLSemantics(parsedResult);
       },
       { priority: "low" },
     );
 
     /**
-     * Hack to transform <math> with LaTeX into $$LaTeX$$ in editor.setData().
+     * Hack to transform <math> with LaTeX into $$LaTeX$$ and formula images in editor.setData().
      */
     editor.data.on(
       "set",
       (e, args) => {
         // Retrieve the data to be set on the CKEditor.
         let modifiedData = args[0];
         // Regex to find all mathml formulas.
-        const regexp = /<math(.*?)<\/math>/gm;
+        const regexp = /(<img\b[^>]*>)|(<math(.*?)<\/math>)/gm;
+        const formulas = [];
+        let formula;
 
         // Get all MathML formulas and store them in an array.
         // Using the conditional operator on data.main because the data parameter has different types depending on:
         //    editor.data.set can be used directly or by the source editing plugin.
         //    With the source editor plugin, data is an object with the key `main` which contains the source code string.
         //    When using the editor.data.set method, the data is a string with the content to be set to the editor.
```

Comment:
This comment is no longer true.
I'd update it to something like "Both `data.set` from the source plugin and console command are taken into account as the data received is an image containing the MathML."

Reply:
Updated! I also added comment about handling mathml